### PR TITLE
fix: support safari TP as a target

### LINKED ISSUE #284: TypeError: Invalid Version: null
I'm getting the above error when trying to run eslint with this plugin.

.eslintrc
```
{
  "env": {
    "browser": true,
    "es6": true,
    "jest": true
  },
  "extends": ["airbnb", "prettier", "prettier/react"],
  "globals": {
    "Atomics": "readonly",
    "SharedArrayBuffer": "readonly"
  },
  "parserOptions": {
    "ecmaFeatures": {
      "jsx": true
    },
    "ecmaVersion": 2018,
    "sourceType": "module"
  },
  "plugins": ["react", "prettier", "react-hooks", "compat"],
  "parser": "babel-eslint",
  "rules": {
    "prettier/prettier": "error",
    "react-hooks/rules-of-hooks": "error",
    "react-hooks/exhaustive-deps": "warn",
    "react/jsx-filename-extension": [1, { "extensions": [".js", ".jsx"] }],
    "import/prefer-default-export": 0,
    "no-shadow": 0,
    "react/forbid-prop-types": 0,
    "no-param-reassign": 0,
    "import/no-extraneous-dependencies": [
      "error",
      {
        "devDependencies": [
          "**/*.test.js",
          "**/*.spec.js",
          "**/*.stories.js",
          "**/*.story.js",
          "setupTests.js"
        ]
      }
    ],
    "react/require-default-props": 0,
    "strict": 0,
    "compat/compat": "warn"
  }
}
```

package.json
```
"scripts: {
    "lint": "eslint \"./src/**/*.{js,jsx}\"",
},
"devDependencies": {
    "eslint": "5.16.0",
    "eslint-config-airbnb": "17.1.1",
    "eslint-config-prettier": "6.1.0",
    "eslint-plugin-compat": "3.1.1",
    "eslint-plugin-import": "2.18.2",
    "eslint-plugin-jsx-a11y": "6.2.3",
    "eslint-plugin-prettier": "3.1.0",
    "eslint-plugin-react": "7.14.3",
    "eslint-plugin-react-hooks": "1.7.0",
}
```

What might this be coming from?

### PR DESCRIPTION
**Pull request by eligao**
closes #284

### CODE REVIEW

File: test/MdnProvider.spec.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import DetermineTargetsFromConfig, { Versioning } from '../src/Versioning';
+import { getUnsupportedTargets } from '../src/providers/MdnProvider';
+
+describe('MdnProvider', () => {
+  it('should support Safari TP', () => {
+    const node = { protoChainId: 'AbortController' };
```
Comment (eligao) [inline test/MdnProvider.spec.js]: Using `AbortController` which would have triggered #284 , but the actual capabilities of Safari TP can change over time, and the test doesn't necessarily work in the future. Any suggestions?

Comment (ljharb) [reply test/MdnProvider.spec.js]: AbortController is in regular Safari, so it's not ever going to disappear from TP.

File: src/providers/MdnProvider.js [hunk 2]
```diff
@@ -79,6 +79,12 @@ export function mdnSupported(node: Node, { version, target }: Target): boolean {
   // If a version is true then it is supported but version is unsure
   if (typeof versionAdded === 'boolean') return versionAdded;
   if (versionAdded === null) return true;
+
+  // Special case for Safari TP: TP is always gte than any other releases
```
Comment (ljharb) [inline src/providers/MdnProvider.js]: let's also handle when `customCoerce(version)` is `null`, so the general problem is solved.

File: test/MdnProvider.spec.js (hunk 1 above)
Comment (eligao) [reply test/MdnProvider.spec.js]: Great.

File: src/providers/MdnProvider.js (hunk 2 above)
Comment (eligao) [reply src/providers/MdnProvider.js]: I thought about doing that but is it safe to assume all cases when customCoerce(version) is null should be treated like `Safari TP`?

Comment (ljharb) [reply src/providers/MdnProvider.js]: no, but i don't want non-safari null cases to crash in the future.

Comment (eligao) [reply src/providers/MdnProvider.js]: Makes sense, updated.

File: src/providers/MdnProvider.js [hunk 3]
```diff
@@ -87,10 +87,14 @@ export function mdnSupported(node: Node, { version, target }: Target): boolean {
...
   // to the first version of the browser that API was added in
-  return semver.gte(
-    semver.coerce(customCoerce(version)),
-    semver.coerce(customCoerce(versionAdded))
-  );
+  const semverCurrent = semver.coerce(customCoerce(version));
+  const semverAdded = semver.coerce(customCoerce(versionAdded));
+
+  // semver.coerce() might be null for non-semvers (other than Safari TP)
+  // Just treat features as supported here for now to avoid lint from crashing
```
Comment (ljharb) [inline src/providers/MdnProvider.js]: we may want to `console.warn` with `version`, though, to encourage people to file issues? (but not for Safari TP, which is known)

Comment (eligao) [reply src/providers/MdnProvider.js]: updated, any suggestions?

**Comment by amilajack** [comment]
Published these changes to eslint-plugin-compat@next

**Comment by eligao** [comment]
> Published these changes to `eslint-plugin-compat@next`. Please let me know if these changes work for you and I will cut a new release for `3.5.1`.

I tested locally on my previously bugged project by using `eslint-plugin-compat@3.5.1-0` and forced `Safari TP` in `.browserslistrc`, and the lint passed successfully.

**Pull request merged**